### PR TITLE
feat: default order for branches and repos

### PR DESCRIPTION
**Pull request by alexnm**
Moved from sorting by name or implicit (for branches) to always sorting by `lastAccessedAt`.

This temporarily improves the experience in the dashboard until we get more advanced filtering/search features. As you can see in the video, the main branch is usually among the ones last visited so it pops up at the top, but without any special treatment, it is ordered like all other branches. Also the most common used repos will be always first, as lastAccessedAt for a repo is actually lastAccessedAt for any of its branches

https://user-images.githubusercontent.com/9945366/209191598-1d80d948-53db-4d79-9956-4333d10f8877.mov

This PR also conditionally renders the link to the branch on GitHub only if that branch exists on remote

### CODE REVIEW

**Review by olarclara — APPROVED** [review]
🔼🔽

File: packages/app/src/app/pages/Dashboard/Content/routes/RepositoryBranches/index.tsx [hunk 1]
```diff
@@ -48,12 +51,13 @@ export const RepositoryBranchesPage = () => {
...
     }
 
-    const branchItems: DashboardGridItem[] = repositoryProject.branches.map(
-      branch => ({
-        type: 'branch',
-        branch,
-      })
-    );
+    const branches = [...repositoryProject.branches];
+    const orderedBranches = branches.sort(sortByLastAccessed);
```
Comment (tristandubbeld) [inline packages/app/src/app/pages/Dashboard/Content/routes/RepositoryBranches/index.tsx:55]: Any particular reason why `branches` is created?

**Review by tristandubbeld — APPROVED** [review]
👏

File: packages/app/src/app/pages/Dashboard/Content/routes/RepositoryBranches/index.tsx (hunk 1 above)
Comment (alexnm) [reply packages/app/src/app/pages/Dashboard/Content/routes/RepositoryBranches/index.tsx:55]: sort mutates the array, and branches is the direct reference of the data coming from overmind/graphql.

**Pull request merged**